react-app/src/components/Extract.js

```diff
@@ -0,0 +1,106 @@
+        }
+
+        this.onConfigFileChange = this.onConfigFileChange.bind;
+        this.onLogFileChange = this.onLogFileChange.bind;
+        this.onIncludeDebugChange = this.onIncludeDebugChange.bind;
+        this.onFilterStartChange = this.onFilterStartChange.bind;
+        this.onFilterEndChange = this.onFilterEndChange.bind;
+        this.onSubmit = this.onSubmit.bind(this);
+    }
+
+    onConfigFileChange(e) {
+        this.setState({
+            configPath: e.target.value
+        });
+    }
+
+    onLogFileChange(e) {
+        this.setState({
+            logPath: e.target.value
+        });
+    }
+
+    onIncludeDebugChange(e) {
+        this.setState({
+            includeDebug: e.target.value
+        });
+    }
+
+    onFilterStartChange(e) {
+        this.setState({
+            filterStart: e.target.value
+        });
+    }
+
+    onFilterEndChange(e) {
+        this.setState({
+            filterEnd: e.target.value
+        });
+    }
+
+    onSubmit(e) {
+        console.log(e);
+        console.log(this.configPath);
+    }
+
+    render() {
+        this.whatever = 10;
+        return <div>
+            <h1 className="page-title">Extract New</h1>
+            <Form>
+                <Row>
+                    <Col>
+                        <Form.Group controlId="formConfigPath" className="mb-3">
+                            <Form.Label className="form-label" >Configuration File</Form.Label>
+                            <Form.Control type="text" />
+                        </Form.Group>
+                        <Form.Group controlId="formLogPath">
+                            <Form.Label className="form-label" >Previous Log File</Form.Label>
+                            <Form.Control type="text" />
+                        </Form.Group>
+                        <Form.Group controlId="formIncludeDebug">
+                            <Form.Check type="checkbox" label="Log output debugging information" />
+                        </Form.Group>
+                    </Col>
+                </Row>
+                <Row>
+                    <Col>
+                        <Form.Group controlId="formIncludeStartDate">
+                            <Form.Check type="checkbox" label="Filter by start date" />
+                        </Form.Group>
+                    </Col>
+                    <Col>
+                        <Form.Group controlId="formIncludeEndDate">
+                            <Form.Check type="checkbox" label="Filter by end date" />
+                        </Form.Group>
+                    </Col>
+                </Row>
+            </Form>
+            <div className="nav-button-container">
+                <LinkContainer to="/">
+                    <button className="nav-button" variant="nav" id="nav-button">Cancel</button>
+                </LinkContainer>
+                <button className="nav-button" type="submit" variant="nav" id="nav-button" onClick={this.onSubmit}>Submit</button>
```

Comment:
With my above comments to get rid of the CDNs and properly install/import bootstrap, you should now be able to use the bootstrap `Button` component and its variants properly.

I remember you mentioning that you intended to override some styles that bootstrap has on their buttons. We can talk about the details of that, but I still think we want to be consistent and use the bootstrap components where possible to avoid the need for more CSS.

Reply:
The main thing that I wanted to override was the color of the button. Bootstrap as already installed before, but I can't use any custom css with the variant option for the buttons because it gets overridden by built-in styles.

Reply:
I have replaced the <button> components with <Button>. I used the "danger" variant for the cancel button and the "info" variant for the submit button.